cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/CreateSegmentCommand.java

```diff
@@ -0,0 +1,91 @@
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.common.Exceptions;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.controller.server.WireCommandFailedException;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import io.pravega.shared.protocol.netty.WireCommands;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Executes a CreateSegment request against the chosen Segment Store instance.
+ */
+public class CreateSegmentCommand extends SegmentStoreCommand {
+
+    /**
+     * Creates a new instance of the CreateSegmentCommand.
+     *
+     * @param args The arguments for the command.
+     */
+    public CreateSegmentCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() {
+        ensureArgCount(2);
+
+        final String fullyQualifiedSegmentName = getArg(0);
+        final String segmentStoreHost = getArg(1);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        ConnectionPool pool = createConnectionPool();
+        @Cleanup
+        Controller controller = instantiateController(pool);
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient, pool);
+
+        String[] inputParam = fullyQualifiedSegmentName.split("/");
+        Preconditions.checkArgument(inputParam.length == 3, "Invalid qualified-segment-name  '%s'", fullyQualifiedSegmentName);
+
+        try {
+            CompletableFuture<WireCommands.StreamSegmentInfo> segmentInfoFuture = segmentHelper.getSegmentInfo(fullyQualifiedSegmentName,
+                    new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken(), 0L);
+            segmentInfoFuture.join();
+            output("Segment %s already exists", fullyQualifiedSegmentName);
+        } catch (Exception ex) {
+            Throwable cause = Exceptions.unwrap(ex);
+            if (cause instanceof WireCommandFailedException && ((WireCommandFailedException) cause).getReason().equals(WireCommandFailedException.Reason.SegmentDoesNotExist)) {
+                StreamConfiguration streamConfig = controller.getStreamConfiguration(inputParam[0], inputParam[1]).join();
+                CompletableFuture<Void> reply = segmentHelper.createSegment(streamConfig.getScalingPolicy(),
+                        fullyQualifiedSegmentName, streamConfig.getRolloverSizeBytes(), super.authHelper.retrieveMasterToken(),
+                        0L,
+                        new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()));
+                reply.join();
+                output("CreateSegment: %s created successfully", fullyQualifiedSegmentName);
+            } else {
+                output("CreateSegment failed for %s - %s", fullyQualifiedSegmentName, ex.getMessage());
```

Comment:
How is this message displayed in case of failures for segmentHelper.createSegment call ( line 74)?

Reply:
In case of failure for segmentHelper.createSegment call , it prints the error as below
`java.util.concurrent.CompletionException: io.pravega.controller.server.WireCommandFailedException: WireCommandFailed with type CREATE_SEGMENT reason ConnectionFailed
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:990)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:970)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at io.pravega.client.connection.impl.ConnectionPoolImpl.lambda$getClientConnection$3(ConnectionPoolImpl.java:159)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:986)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:970)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at io.pravega.common.concurrent.ThreadPoolScheduledExecutorService$ScheduledRunnable.run(ThreadPoolScheduledExecutorService.java:197)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.pravega.controller.server.WireCommandFailedException: WireCommandFailed with type CREATE_SEGMENT reason ConnectionFailed
	at io.pravega.controller.server.SegmentHelper.processAndRethrowException(SegmentHelper.java:843)
	at io.pravega.controller.server.SegmentHelper.lambda$sendRequest$28(SegmentHelper.java:832)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:986)
	... 13 more`

Reply:
covered the segmentHelper.createSegment call inside try catch block.